packages/unified-selection/api/unified-selection.api.md

```diff
@@ -36,9 +82,27 @@ export interface CustomSelectable {
     loadInstanceKeys: () => AsyncIterableIterator<SelectableInstanceKey>;
 }
 
+// @internal
+interface ElementSelectionScopeProps {
+    ancestorLevel?: number;
+    id: "element" | "functional";
+}
+
 // @beta
 export function enableUnifiedSelectionSyncWithIModel(props: EnableUnifiedSelectionSyncWithIModelProps): () => void;
 
+// @internal
+interface EnableUnifiedSelectionSyncWithIModelProps {
```

Comment:
Here we have an `@internal` API exposed through a `@beta` one. Two questions:
1. Why doesn't `check-internal` script complain about `@internal` being exposed here in API report?
2. Why doesn't `extract-api` script complain about public API exposing internal one?

Reply:
> Why doesn't check-internal script complain about @internal being exposed here in API report?

I couldn't figure out a simple way to exclude this check only for APIs, exported from internal fields so I removed it for APIs that are not exported through barrel (no `export` keyword in API.md). This was needed because we have at least one instance of `@internal` interface member (in presentation-components
`ViewWithUnifiedSelectionProps.selectionHandler`). Judging from this comment in `check-internal` script I figured that this is something that we have to account for:
```
// only look for non-indented tags which target root level APIs - we still want to allow things
// like @internal class functions and interface attributes
```
In this particular case it looks like I missed this after disabling the internal check.

> Why doesn't extract-api script complain about public API exposing internal one?

It checks if release tags are "compatible" and not necessarily equal. This is supposed to guard against cases where creation of `public` objects would require to use `beta`, `alpha` or `internal` APIs. See [docs about this](https://api-extractor.com/pages/messages/ae-incompatible-release-tags/). Although, I am not sure why it allows for this. Maybe it handles unexported types in a different way, or maybe this is a bug. Unexported APIs is something that the authors are very reluctant to support or document about (they call this "forgotten exports").

Reply:
Added [another backlog item](https://github.com/iTwin/presentation/issues/652) to fix occurrences of `@internal` in the public API.